File: React.UI/ClientApp/src/models/Pago.ts
import { alumnoSchema } from "./Alumno";
import { conceptoSchema } from "./Concepto";
import { tutorSchema } from "./Tutor";
import { z } from "zod";
import {numeric, text, id, file} from "./Schema";

const pagoSchema = z.object({
  idPago: id,
  fecha: z.date().default(new Date()),
  cantidad: numeric,
  concepto: numeric,
  conceptoNavigation: conceptoSchema.nullable(),
  comprobanteDePago: file,
  formatoComprobante: text,
  idTutor: numeric,
  idTutorNavigation: tutorSchema.nullable(),
  idAlumno: numeric,
  idAlumnoNavigation: alumnoSchema.nullable(),
});

type IPago = z.TypeOf<typeof pagoSchema>;

export type { IPago };

export { pagoSchema };

File: Academic.API/Models/Kinder/Pago.cs
using System;
using System.Collections.Generic;

#nullable disable
namespace Academic.API.Models.Kinder
{
    public partial class Pago
    {
        public int IdPago { get; set; }
        public DateTime Fecha { get; set; }
        public int Concepto { get; set; }
        public byte[] ComprobanteDePago { get; set; }
        public int IdTutor { get; set; }
        public int IdAlumno { get; set; }
        public int Cantidad { get; set; }
        public string FormatoComprobante { get; set; }

        public virtual ConceptosDePago ConceptoNavigation { get; set; }
        public virtual Alumno IdAlumnoNavigation { get; set; }
        public virtual Tutor IdTutorNavigation { get; set; }
    }
}
